unlock-app/src/components/creator/CreatorLockForm.js

```diff
@@ -34,11 +34,20 @@ class CreatorLockForm extends React.Component {
     this.handleChange = this.handleChange.bind(this)
   }
 
+  validate(name, value) {
+    return (name === 'name' && typeof value === 'string' && value.length > 0) ||
+      ((name === 'expirationDuration' || name === 'maxNumberOfKeys') & !isNaN(value) && value > 0) ||
+      (name === 'keyPrice' && !isNaN(value) && value >= 0)
+  }
+
   handleChange(event) {
+    event.target.dataset.valid = this.validate(event.target.name, event.target.value)
     this.setState({ [event.target.name]: event.target.value })
   }
 
   handleSubmit() {
+    if (document.querySelector('[data-valid="false"]')) return false
```

Comment:
I think we could maybe even show an error message in the error panel here later (I worry that just showing an error state may not *tell* what the error actually is)? But @smombartz can weigh on this!